### DISCUSSION TITLE
Update cache by calling a query

### DISCUSSION THREAD

**Discussion by gaspoute** [urql-graphql/urql#2079] [Q&A] [answered]
Hello,

Is there a way to call a query to update the cache ?

Let me explain. I have a list of budget lines and a list of activities. Each activity is related to a budget line. When I create an activity, I want to update only the budget line related to this new activity. I have a query in the schema that allows me to do this, but this query is not used anywhere else. It was added only to update the cache (at least for now).

Thanks :pray: 
Gaspar

**Comment by JoviDeCroock** [comment, marked as answer]
You can leverage our [normalized cache](https://formidable.com/open-source/urql/docs/graphcache/) [update functions](https://formidable.com/open-source/urql/docs/graphcache/) for that. That's assuming you create that item with help of a mutation

**Reply by gaspoute** [reply]
That's what I do. I invalidate the budget line that is related to the new activity. After that, the list of budget lines is refetched. But that's what I would like to avoid. I would like to update the budget line using a query that retrieves the one and not the entire list. This query is available but not used in the application, therefore unknown to the cache. I read the doc several times but I don't understand how to do it, I must have missed something.

**Reply by JoviDeCroock** [reply]
I don't know how your data is structured but both `writeFragment` and `updateQuery` could be used for this. Let's say your typename is `BudgetLine` then we can do:

```js
writeFragment(
  gql`
    fragment x on BudgetLine { id price }
  `, { id: 'your-id-for-the-line-you-want-to-update', price: newPrice }
 )
```

This would be instead of using `invalidate`
